projects/vdk-plugins/vdk-impala/tests/template_regression_test.py

```diff
@@ -0,0 +1,582 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import os
+import pathlib
+import re
+import time
+import unittest
+from unittest.mock import ANY
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+import pytest
+from vdk.internal.core import errors
+from vdk.plugin.impala import impala_plugin
+from vdk.plugin.test_utils.util_funcs import CliEntryBasedTestRunner
+from vdk.plugin.test_utils.util_funcs import get_test_job_path
+
+VDK_DB_DEFAULT_TYPE = "VDK_DB_DEFAULT_TYPE"
+VDK_IMPALA_HOST = "VDK_IMPALA_HOST"
+VDK_IMPALA_PORT = "VDK_IMPALA_PORT"
+
+
+@pytest.mark.skip(
+    reason="We need to test this with a recent impala instance. Current test instance is too old"
+)
+@patch.dict(
+    os.environ,
+    {
+        VDK_DB_DEFAULT_TYPE: "IMPALA",
+        VDK_IMPALA_HOST: "localhost",
+        VDK_IMPALA_PORT: "21050",
+    },
+)
+class TemplateRegressionTests(unittest.TestCase):
+    def setUp(self) -> None:
+        self.__runner = CliEntryBasedTestRunner(impala_plugin)
+        time.sleep(10)  # wait for impala instance to come online
+
```

Comment:
The issue was, there wasn't **vdkprototypes** database provisioned, for example:
```
-- job_name: load_versioned_template_only
-- op_id: ed243969-d0a6-4b51-aa7c-81c51028df7e-1642665697
DESCRIBE formatted `vdkprototypes`.`dim_sddc_h_as_textfile`
2022-01-20 10:01:37,525 [VDK] load_versioned_template_only [INFO ] vdk.plugin.impala.impala_error impala_error_handler:72   handle_error    [id:ed243969-d0a6-4b51-aa7c-81c51028df7e-1642665697-7f222]- Try (1 of 5) to handle exception AnalysisException: Database does not exist: vdkprototypes
2022-01-20 10:01:37,525 [VDK] load_versioned_template_only [ERROR] vdk.plugin.impala.impala_conne            errors.py:373  log_and_rethrow [id:ed243969-d0a6-4b51-aa7c-81c51028df7e-1642665697-7f222]- An error in data job code  occurred. The error should be resolved by User Error. Here are the details:
  WHAT HAPPENED : Executing query FAILED.
WHY IT HAPPENED : An exception occurred, exception message was: AnalysisException: Database does not exist: vdkprototypes
```

A solution would be to add to `setUp()`:
`self._run_query("CREATE DATABASE IF NOT EXISTS vdkprototypes")`

In that sense, all of the other tests were also using vdkprototypes, yet not failing. This is due to such implementation:
```
actual_rs = self._run_query(f"SELECT * FROM {test_schema}.{target_table}")
expected_rs = self._run_query(f"SELECT * FROM {test_schema}.{source_view}")
assert actual_rs.output == expected_rs.output
```
That relies the expected_rs is in fact operational, without explicitly checking the results status.
Solution is, to add explicit status check to each test that is missing such.

Reply:
I did try adding  self._run_query("CREATE DATABASE IF NOT EXISTS vdkprototypes") initially, however the tests are still failing. I did try to make the scd1 template working, however after a few tweaks to the sql code I reach a point where running the following query: `REFRESH {target_schema}.{target_table};
` bugs the docker instance of impala and every other subsequent query fails with an error message along the lines of `Failed after retrying 3 times`. Even SQL queries that previously succeeded fail with the same message after running the refresh query. That's why I think we can't really test templates against the test impala version present.

Reply:
All tests are working on my local machine;
could you update the PR and see what's going on with the CI/CD?

Reply:
Latest commit running against the docker impala has 8 tests failing and 8 passing for me. I've added a commit to check the CI/CD runner too.

Reply:
I see; find https://github.com/vmware/versatile-data-kit/pull/675 that provides an Impala image, compatible with the present tests